wd_core.au3

```diff
@@ -1675,20 +1676,25 @@ EndFunc   ;==>__WD_CloseDriver
 ; #INTERNAL_USE_ONLY# ===========================================================================================================
 ; Name ..........: __WD_EscapeString
 ; Description ...: Escapes designated characters in string.
-; Syntax ........: __WD_EscapeString($sData)
-; Parameters ....: $sData - the string to be escaped
-; Return values..: Escaped string.
+; Syntax ........: __WD_EscapeString($sData[, $iOption = 0])
+; Parameters ....: $sData   - the string to be escaped
+;                  $iOption - [optional] Default is 0.
+; Return values .: Escaped string.
 ; Author ........: Danp2
 ; Modified ......:
-; Remarks .......:
+; Remarks .......: See $JSON_* constants in json.au3 for $iOption possibilities
 ; Related .......:
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func __WD_EscapeString($sData)
-	Local $sRegEx = "([" & $_WD_ESCAPE_CHARS & "])"
-	Local $sEscaped = StringRegExpReplace($sData, $sRegEx, "\\$1")
-	Return SetError($_WD_ERROR_Success, 0, $sEscaped)
+Func __WD_EscapeString($sData, $iOption = 0)
+	If BitAND($iOption, $JSON_MLREFORMAT) Then
+		$sData = StringRegExpReplace($sData, '(?m)[\R\t]','') ; Strip tabs and CR/LFs
+	Endif
+
+	$sData = Json_StringEncode($sData, $iOption) ; Escape JSON Strings
```

Comment:
`$JSON_MLREFORMAT`  is used only in `_WD_ExecuteScript()`
previously it was `__WD_EscapeString($sScript)` so we should add `$iOption = 0`
like this:
```autoit
Func __WD_EscapeString($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '[\r\t]','')
		$iOption = 0
	Endif

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString
```

Reply:
I'm not sure why you would want to do this. The use of `$JSON_MLREFORMAT` is designed so that it doesn't impact  `Json_StringEncode`. Your method would prevent doing something like this --

```autoit
$sStr = '"this is ' & @CRLF & 'a test"'
$sResult = __WD_EscapeString($sStr, $JSON_HEX_QUOT + $JSON_MLREFORMAT)
```

Reply:
You mean that after $JSON_MLREFORMAT is performed the $iOption is like a fake option because all other `true` bits will be masked to 0 ?

Reply:
The assumption is that the C code is performing some form of bitwise operation to check each of the lower 10 bits to determine if they are on or off. Therefore, setting a bit on that is above the first 10 shouldn't affect to operation of the C code.

If you wanted to be sure, a better solution would be to do something like this --

`$iOption = BitXOR($iOption, $JSON_MLREFORMAT)`

Reply:
> $iOption = BitXOR($iOption, $JSON_MLREFORMAT)

Do this.

Reply:
also header
```autoit
; #INTERNAL_USE_ONLY# ===========================================================================================================
; Name ..........: __WD_EscapeString
; Description ...: Escapes designated characters in string to fullfill "WebDriver" requirements
; Syntax ........: __WD_EscapeString($sData[, $iOption = 0])
; Parameters ....: $sData               - JavaScript or JSON string
;                  $iOption             - One of $JSON_* constants or $JSON_MLREFORMAT. Default is 0.
; Return values .: escaped string
; Author ........: Danp2
; Modified ......: mLipok
; Remarks .......:
; Related .......:
; Link ..........:
; Example .......: No
; ===============================================================================================================================
Func __WD_EscapeString($sData, $iOption = 0)
	If BitAND($iOption, $JSON_MLREFORMAT) Then ; JavaScript linearization first
		$sData = StringRegExpReplace($sData, '[\r\t]','')
		$iOption = BitXOR($iOption, $JSON_MLREFORMAT)
	Endif

	$sData = Json_StringEncode($sData, $iOption) ; JSON Escape Strings

	Return SetError($_WD_ERROR_Success, 0, $sData)
EndFunc   ;==>__WD_EscapeString
```

Reply:
> > $iOption = BitXOR($iOption, $JSON_MLREFORMAT)
> 
> Do this.

I can, but it isn't needed IMO.